### PR TITLE
Fix UI usage

### PR DESCRIPTION
**Pull request by floitsch**
The 'profile show' was just stringifying the map.
The `--output-format` ... flags didn't work correctly.

### CODE REVIEW

File: src/cli/cmds/serial.toit [hunk 1]
```diff
@@ -258,7 +258,7 @@ flash invocation/Invocation:
             --cli=cli
 
       if ui.wants-structured --kind=Ui.RESULT:
-        ui.emit --result {
+        ui.emit-map --result {
               "device_id": "$device-id",
```
Comment (kasperl) [inline src/cli/cmds/serial.toit:262]: That's a lot of indentation?

File: src/cli/cli.toit [hunk 2]
```diff
@@ -57,22 +57,8 @@ main args --cli/Cli?:
         Option "fleet"
             --type="directory|reference"
             --help="Specify the fleet. Can also be set with the ARTEMIS_FLEET environment variable.",
-        OptionEnum "output-format"
```
Comment (kasperl) [inline src/cli/cli.toit:60]: You new got these for free if you don't have options with these names, right?

File: tests/cmd-device-extract-test.toit [hunk 3]
```diff
@@ -19,7 +19,7 @@ main args:
 run-test fleet/TestFleet --args/List:
   with-tmp-directory: | tmp-dir |
     result := fleet.tester.run --json [
-      "fleet", "add-device", "--output-format", "json",
```
Comment (kasperl) [inline tests/cmd-device-extract-test.toit:22]: This is just handled through the `--json` argument, right?

File: tests/cmd-fleet-add-device-test.toit [hunk 4]
```diff
@@ -63,7 +63,7 @@ run-test fleet/TestFleet --args/List:
 
     // Get the device info as json.
     result := fleet.run --json [
-      "fleet", "add-device", "--format", "tar", "-o", tar-file, "--output-format", "json"
```
Comment (kasperl) [inline tests/cmd-fleet-add-device-test.toit:66]: This is just handled through the `--json` argument, right?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/cli/cmds/serial.toit (hunk 1 above)
Comment (floitsch) [reply src/cli/cmds/serial.toit]: done.

**Pull request merged**